packages/astro/src/@types/astro.ts

```diff
@@ -1375,6 +1375,75 @@ export interface AstroUserConfig {
 		 * ```
 		 */
 		devOverlay?: boolean;
+
+		// TODO review with docs team before merging to `main`
+		/**
+		 * @docs
+		 * @name experimental.i18n
+		 * @type {object}
+		 * @version 3.*.*
+		 * @type {object}
+		 * @description
+		 *
+		 * Configures experimental i18n routing and allows you to specify some customization options.
+		 */
+		i18n?: {
+			/**
+			 * @docs
+			 * @name experimental.i18n.defaultLocale
+			 * @type {string}
+			 * @version 3.*.*
+			 * @description
+			 *
+			 * The default locale of your website/application
+			 */
+			defaultLocale: string;
```

Comment:
Could we make it a generic so that we can enforce it in `locales` at the type level?

Reply:
We were planning on adding better type support through some type generation.

Reply:
yeah codegen is probably easier than generics in this case